source/guides/faq.md

```diff
@@ -49,6 +49,44 @@ See <https://github.com/nix-community/home-manager>
 
 ## NixOS
 
+### How to run non-nix executables?
+
+NixOS cannot run dynamically linked executables intended for generic linux environments out of the box.
+This is because, by design, it does not have a global library path.
+Solutions include:
+
+- Use the version packaged in nixpkgs, if there is one.
+  You can search available packages at <https://search.nixos.org/packages>.
+
+- Write a nix expression for the program to package it in your own configuration.
+  There are multiple approaches:
+  - Build from source.
+    For an introduction to this, see [](../tutorials/packaging-existing-software).
+  - Set the program's ELF metadata to include paths to libraries using [`autoPatchelfHook`](https://nixos.org/manual/nixpkgs/stable/#setup-hook-autopatchelfhook).
+    Do this if building from source isn't feasible.
+  - Wrap the program to run in a normal-seeming container using [`buildFHSEnv`](https://nixos.org/manual/nixpkgs/stable/#sec-fhs-environments).
+    This is a last resort, but sometimes necessary.
+    If the program downloads and runs other executables, for example.
+
+- Create a library path that only applies to unpackaged programs by using `nix-ld`.
+  Note that this does not work for 32-bit executables on `x86_64` machines.
+  Add this to your `configuration.nix`, rebuild, and relogin to propagate the new environment variables:
+
+  ```nix
+  programs.nix-ld.enable = true;
+  programs.nix-ld.libraries = with pkgs; [
+    # Add any missing dynamic libraries for unpackaged programs
+    # here, NOT in environment.systemPackages
+  ];
+  ```
```

Comment:
~~~suggestion
  ```
  
  Then log out and back in again.
~~~

A bit harder to miss this way.

Also why is this required at all? Ideally this wouldn't be necessary.

Reply:
Reworked that so the directions for after editing `configuration.nix` are after the code block.